docs/API.md

```diff
@@ -675,6 +676,81 @@ exports[`drinking flavors throws on octopus 1`] = `"yuck, octopus flavor"`;
 
 Check out [React Tree Snapshot Testing](http://facebook.github.io/jest/blog/2016/07/27/jest-14.html) for more information on snapshot testing.
 
+
+### Extending Jest Matchers
+
+Using descriptive matchers will help your tests be readable and maintainable. Jest has a simple API for adding your own matchers. Here is an example of adding a matcher:
+
+```js
+expect.extend({
+  toContainLetter(received, expected) {
+    // some defensive programming
+    if (this.utils.getType(received) !== 'string') {
+      throw new Error(
+        this.utils.matcherHint('[.not].toContainLetter') + '\n\n' +
+        `Received value must be a string.\n` +
+        printWithType('Received', received, this.utils.printReceived),
+      );
+    }
+
+    // some defensive programming
+    if (this.utils.getType(expected) !== 'string' || expected.length > 1) {
+      throw new Error(
+        this.utils.matcherHint('[.not].toContainLetter') + '\n\n' +
+        `Expected value must be a single letter string.\n` +
+        printWithType('Expected', expected, this.utils.printExpected),
+      );
+    }
+
+    // assertion
+    const pass = received.split('').some(letter => letter === expected);
+
+
+    // message formatting
+    const message = pass
+      ? () => matcherHint('.not.toContainLetter') + '\n\n' +
+        `Expected value to not contain the letter:\n` +
+        `  ${printExpected(expected)}\n` +
+        `Received:\n` +
+        `  ${printReceived(received)}`
+      : () => matcherHint('.toContainLetter') + '\n\n' +
+        `Expected value to contain the letter:\n` +
+        `  ${printExpected(expected)}\n` +
+        `Received:\n` +
+        `  ${printReceived(received)}`
+
+    return { message, pass };
+  }
+});
+
+describe('toContainLetter', () => {
+  it('throws when given the wrong arguments', () => {
+    expect(() => expect(1).toContainLetter('l')).toThrow();
+    expect(() => expect('word').toContainLetter()).toThrow();
+  });
+
+  it('matchers letters inside a word', () => {
+    expect('Jest').toContainLetter('J');
+    expect('Jest').not.toContainLetter('m');
+  });
+});
+```
+
+Jest will give your matchers context to simplify coding. The following can be found on `this` inside a custom matcher:
+
+#### `this.dontThrow`
```

Comment:
`dontThrow()` is pretty useless on its own :)
can you add that instead of throwing, it saves them to the global `state.supressedErrors` that you can access via `getState` and `setState`?

example: https://github.com/facebook/jest/blob/master/packages/jest-jasmine2/src/setup-jest-globals.js#L15